lib/aurora_bootstrap_parallelization/cronjob_exporter.rb

```diff
@@ -0,0 +1,25 @@
+require 'yaml'
+require 'erb'
+
+module AuroraBootstrapParallelization
+    # for scheduled run of aurora-bootstrap script
+    class CronjobExporter < Exporter
+      def initialize( config, is_cron_job )
+        super(config, true)
+      end
+
+      def manifest
```

Comment:
Can we share this code, since it's basically identical? Like

```ruby
module AuroraBootstrapParallelization
  class Exporter
    def initialize( config )
      @name    = config[ "name" ]
      @env     = config[ "env" ]
      @version = config[ "version" ]
    end

    def deploy
      write_manifest
      kubectl_deploy
    end

    def manifest
      @manifest ||= begin
        erb = ERB.new( template_path ).result( binding )
        YAML.load( erb ).tap do | manifest |
          manifest.dig( *env_path )[0][ "env" ] = @env
        end
      end
    end

    def template_path
      File.expand_path( "../../../templates/#{TYPE}.yml", __FILE__ )
    end

    def file_name
      @file_name ||= "#{@name}-#{TYPE}.yml"
    end
      
    def kubectl_deploy
      spawn "kubectl -n aurora-bootstrap create -f /tmp/#{file_name}"
    end

    def write_manifest
      file = File.open("/tmp/#{file_name}", 'w')
      file.write manifest.to_yaml
      file.close
    end
  end

  class CronJobExporter < Exporter
   TYPE = :cronjob

    def env_path
      ["spec", "jobTemplate", "spec", "template", "spec", "containers"]
   end
  end

  class JobExporter < Exporter
   TYPE = :job

    def env_path
      [ "spec", "template", "spec", "containers" ]
    end
  end
end
```

Does this make sense?

Reply:
Nice

Reply:
NoMethodError: undefined method `to_hash' for #<String:0x00007fcf2411eae0>. Looks like manifest is a string and dig method cannot be used.

Reply:
There is some issue with the way you defined TYPE

Reply:
I changed to using instance var instead

Reply:
Oh. I think the base class would have to define it and the inherited can override it. instance works just as well though